packages/app/src/provider/BackgroundLockProvider.tsx

```diff
@@ -0,0 +1,39 @@
+import type { PropsWithChildren } from 'react'
+
+import { useSecureUnlock } from '@package/secure-store/secure-wallet-key/SecureUnlockProvider'
+import { useEffect, useRef } from 'react'
+import React from 'react'
+import { AppState, type AppStateStatus } from 'react-native'
+
+const BACKGROUND_TIME_THRESHOLD = 3000 // FIXME
+
+export function BackgroundLockProvider({ children }: PropsWithChildren) {
+  const secureUnlock = useSecureUnlock()
+  const backgroundTimeRef = useRef<Date | null>(null)
+
+  useEffect(() => {
+    const handleAppStateChange = (nextAppState: AppStateStatus) => {
+      if (nextAppState === 'background' || nextAppState === 'inactive') {
+        backgroundTimeRef.current = new Date()
+      } else if (nextAppState === 'active') {
+        if (backgroundTimeRef.current) {
+          const timeInBackground = new Date().getTime() - backgroundTimeRef.current.getTime()
```

Comment:
Shouldn't we already lock it in the background? 
So if th app becomes inactive /background we `setTimeout`. once it becomes active we clear timeout. if timeout is executed we lock the wallet